map_server/src/main.cpp

```diff
@@ -224,28 +264,36 @@ class MapServer
 int main(int argc, char **argv)
 {
   ros::init(argc, argv, "map_server", ros::init_options::AnonymousName);
-  if(argc != 3 && argc != 2)
+  if(argc > 3)
```

Comment:
I would suggest to refactor this to a switch statement to improve readability, it is c++ after all, and to add some comment/documentation for each case.